tests/unit/lms/services/lti_h_test.py

```diff
@@ -14,30 +15,68 @@ def test_sync_does_nothing_if_provisioning_is_disabled(
     ):
         ai_getter.provisioning_enabled.return_value = False
 
-        lti_h_svc.sync(groups=[h_group])
+        lti_h_svc.sync(h_groups=[h_group])
 
-        h_api.upsert_user.assert_not_called()
+        h_api.bulk_action.assert_not_called()
 
-    def test_sync_catches_HAPIErrors(self, h_api, lti_h_svc, h_group):
-        h_api.upsert_user.side_effect = HAPIError
+    def test_sync_catches_HAPIErrors(
+        self, h_api, lti_h_svc, h_group, group_info_service
+    ):
+        h_api.bulk_action.side_effect = HAPIError
 
         with pytest.raises(HTTPInternalServerError):
-            lti_h_svc.sync(groups=[h_group])
-
-
-class TestGroupUpdating:
-    def test_it_upserts_the_group(self, h_api, lti_h_svc, h_group):
-        lti_h_svc.sync([h_group])
+            lti_h_svc.sync(h_groups=[h_group])
 
-        h_api.upsert_group.assert_called_once_with(h_group)
-
-    def test_it_raises_if_upserting_the_group_fails(self, h_api, lti_h_svc, h_group):
-        h_api.upsert_group.side_effect = HAPIError("Oops")
-
-        with pytest.raises(HTTPInternalServerError, match="Oops"):
-            lti_h_svc.sync([h_group])
+        group_info_service.assert_not_called()
 
-    def test_it_upserts_the_GroupInfo_into_the_db(
+    def test_sync_calls_bulk_action_correctly(self, h_api, h_user, lti_h_svc):
+        groups = [
+            HGroup(
+                name=f"name_{i}", authority_provided_id="test_authority_provided_id",
+            )
+            for i in range(2)
+        ]
+
+        lti_h_svc.sync(h_groups=groups)
+
+        _, kwargs = h_api.bulk_action.call_args
+
+        assert "commands" in kwargs
+
+        # pylint: disable=protected-access
+        assert [command.raw for command in kwargs["commands"]] == [
+            CommandBuilder.user.upsert(
+                {
+                    "authority": lti_h_svc._authority,
+                    "username": h_user.username,
+                    "display_name": h_user.display_name,
+                    "identities": [
+                        {
+                            "provider": h_user.provider,
+                            "provider_unique_id": h_user.provider_unique_id,
+                        }
+                    ],
+                },
+                "user_0",
+            ).raw
+        ] + [
+            CommandBuilder.group.upsert(
+                {
+                    "authority": lti_h_svc._authority,
+                    "name": group.name,
+                    "authority_provided_id": group.authority_provided_id,
+                },
+                f"group_{i}",
+            ).raw
+            for i, group in enumerate(groups)
```

Comment:
```suggestion
            CommandBuilder.group.upsert(
                {
                    "authority": lti_h_svc._authority,
                    "name": group.name,
                    "authority_provided_id": group.authority_provided_id,
                },
                f"group_{i}",
            ).raw
            for i, group in enumerate(groups)
```

Reply:
I'm going blind. I can't spot the difference here? Help a brother out?

Reply:
I think this was a mistake, ignore this